### ISSUE TITLE
Fix CBC access denied error

### ISSUE THREAD

**Issue by palewire** [palewire/news-homepages#124]
![cbcnews](https://user-images.githubusercontent.com/9993/176911713-0aa9c417-991a-417f-b1fb-b67af2b80aa8.jpg)

**Comment by ChrisArchitect** [comment]
(continuing convo from #105) Bummer about these Access Denied's and think it is definitely a User-agent issue common to any scraper activities, probably experienced by many GitHub Actions using Chromium, not just us.

The question is how to change it.

Looking at logs of Actions running, I don't quite understand everything but it seems to install Chromium and then run a bunch of stuff with an extension, ad blocker etc, eventually capturing the screen of the loaded site?

Is there code somewhere that controls that whole process?

Assume it is running Headless Chrome, which I guess many news sites have decided to block on account of bot activity.

If we can add a `page.setUserAgent()` somewhere to set the user-agent for normal Headed Chrome (whatever current version string... 103 being latest...  `Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36` then maybe it will work.

Will examine code/poke around further to see if can figure out on my own.

**Comment by ChrisArchitect** [comment]
Right, replace references to Chromium above with 'Playwright', the testing/engine we're using to generate the shots.

So I see in https://github.com/palewire/news-homepages/blob/78f88c5ec12ce927fb208d9d9642ab2f7a28817c/newshomepages/screenshot.py#L80 there's a `Headless=False` and a `UserAgent` definition.... Is the headless false setting working correctly? And is the UserAgent being a unknown NewsHomepages string maybe throwing off the website security? Maybe try changing it to established Chrome 103 one because even no site is going to recognize NewsHomepages as a valid bot (yet).

**Comment by palewire** [comment]
The headless does appear to be working, it's requirement to get plugins like AdGuard to work with Playwright. I'm not a Chrome version expert, but we are using the default version being installed by Playwright, which I believe is the latest "stable" release so I'm reluctant to change that.

User agent experimentation seems totally possible. I just don't have an opinion about the proper technique to use.

**Comment by ChrisArchitect** [comment]
Right fairplay about letting headless/default do it's thing -- is it maybe the override of of `user_agent` there setting it to News Homepages that might be throwing off some sites? It's tough because I totally understand the site ops deciding to block headless_chrome agents but obv. that's probs for us.

**Comment by palewire** [comment]
We could try a user agent spoofer?

**Comment by ChrisArchitect** [comment]
well yeah, that's what the `user_agent` option is probably for. So we keep headless=false, and then specify a `user_agent` of `Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36` -- doesn't hurt to try that?

**Comment by palewire** [comment]
Hit me with a PR and add the CBC row back into the sites.csv. I think that would do it.

**Comment by ChrisArchitect** [comment]
Will do. Afraid of breaking, but either will work or be harmless in theory. 

Notice CBCnews is still included in Sources, so nothing to do there. Thx!🤞

**Comment by ChrisArchitect** [comment]
ok! Actions is so crazy, over my head the system you've rigged up there, but I can see the artifacts at least! Hello CBC!🎉

**Comment by AsaiToshiya** [comment]
Probably solve #105 and #112 as well, so 4d61358 can be reverted.
